pages/SelectServer.vue

```diff
@@ -28,8 +39,22 @@ export default Vue.extend({
       title: this.$store.state.page.title
     };
   },
+  computed: {
+    ...mapState('servers', ['serverList'])
+  },
+  watch: {
+    serverList(): void {
+      if (this.serverList.length === 0) {
+        this.$router.push('/addserver');
+      }
+    }
+  },
   created() {
-    this.setPageTitle({ title: this.$t('selectServer') });
+    if (!this.serverList.length) {
```

Comment:
This should probably be done in `asyncData` so we check even before rendering the page. You can use `context.redirect()` to redirect from there.